### PR TITLE
List asb & asq transport features that are not compatible with SC

### LINKED ISSUE #2850: Feedback: 'Brokered Message Creation'
Connects to https://github.com/Particular/NServiceBus.AzureServiceBus/issues/572

Feedback for 'Brokered Message Creation' https://docs.particular.net/nservicebus/azure-service-bus/brokered-message-creation

Location in GitHub: https://github.com/Particular/docs.particular.net/blob/master/nservicebus/azure-service-bus/brokered-message-creation.md

ping @Particular/azure-maintainers

The note about `SupportedBrokeredMessageBodyTypes.Stream` says "not compatible with SC", but doesn't mention that a [SC adapter](https://docs.particular.net/servicecontrol/transport-adapter/) can provide a solution meanwhile.

### LINKED ISSUE #2851: Feedback: 'Topologies'
Connects to https://github.com/Particular/NServiceBus.AzureServiceBus/issues/572

Feedback for 'Topologies' https://docs.particular.net/nservicebus/azure-service-bus/topologies/

Location in GitHub: https://github.com/Particular/docs.particular.net/blob/master/nservicebus/azure-service-bus/topologies/index.md

ping @Particular/azure-maintainers 

~~The note about `ForwardingTopology` says "not compatible with SC", but doesn't mention that a [SC adapter](https://docs.particular.net/servicecontrol/transport-adapter/) can provide a solution meanwhile.~~ Topology shouldn't matter at all...

### PR DESCRIPTION
**Pull request by yvesgoeleven**
Fixes #2851 
Fixes #2850 

Connects to https://github.com/Particular/NServiceBus.AzureServiceBus/issues/572

Listed the features
In the description of each feature, explain that it is incompatible & refer to the transport adapter as solution

### CODE REVIEW

**Comment by yvesgoeleven** [comment]
@Particular/azure-maintainers please review

File: nservicebus/azure-storage-queues/multi-storageaccount-support_aliases_asq_[7,].partial.md [hunk 1]
```diff
@@ -21,4 +21,6 @@ snippet: AzureStorageQueueUseMultipleAccountAliasesInsteadOfConnectionStrings2
 
 Aliases can be provided for both the endpoint's connection string as well as other accounts' connection strings. This enables using `@` notation for destination addresses `queue_name@accountName`.
 
-NOTE: The examples above use different default accounts' aliases to enable a coherent addressing. Using the same name, like `default`, for all main accounts is highly discouraged as it introduces ambiguity in resolving addresses like `queue@default`. For example, when an address is interpreted as a reply address, the name `default` will point to a different connection string.
\ No newline at end of file
+NOTE: The examples above use different default accounts' aliases to enable a coherent addressing. Using the same name, like `default`, for all main accounts is highly discouraged as it introduces ambiguity in resolving addresses like `queue@default`. For example, when an address is interpreted as a reply address, the name `default` will point to a different connection string.
```
Comment (SeanFeldman) [inline nservicebus/azure-storage-queues/multi-storageaccount-support_aliases_asq_[7,].partial.md]: Replace "like" with "such as"?

File: servicecontrol/transport-adapter/incompatible-features.md [hunk 2]
```diff
@@ -14,3 +14,15 @@ The [multi-instance](/nservicebus/sqlserver/deployment-options.md#modes-overview
 ## RabbitMQ transport
 
 Neither [direct topology](/nservicebus/rabbitmq/routing-topology.md#direct-routing-topology) nor [custom topologies](/nservicebus/rabbitmq/routing-topology.md#custom-routing-topology) are supported as ServiceControl is configured to use the default topology which expects an exchange exists with the name same as the destination endpoint's name.
+
+## Azure Service Bus
+
+* Using [aliases](/nservicebus/azure-service-bus/securing-connection-strings.md) instead of connectionstring will make it impossible to retry messages from service control.
```
Comment (SeanFeldman) [inline servicecontrol/transport-adapter/incompatible-features.md]: "service control" capitalize

File: servicecontrol/transport-adapter/incompatible-features.md [hunk 3]
```diff
@@ -14,3 +14,15 @@ The [multi-instance](/nservicebus/sqlserver/deployment-options.md#modes-overview
...
+* Leveraging [multiple namespaces in a topology](/nservicebus/azure-service-bus/multiple-namespaces-support.md) will require a setup of multiple service control instances.
+* Customizing [brokered message creation](/nservicebus/azure-service-bus/brokered-message-creation.md) may lead to incompatible wire formats and deserialization errors.
```
Comment (SeanFeldman) [inline servicecontrol/transport-adapter/incompatible-features.md]: May or will? 🙂

File: servicecontrol/transport-adapter/incompatible-features.md [hunk 4]
```diff
@@ -14,3 +14,15 @@ The [multi-instance](/nservicebus/sqlserver/deployment-options.md#modes-overview
...
+
+## Azure Storage Queues
+
+* Using [aliases](/nservicebus/azure-storage-queues/configuration.md#connection-strings-using-aliases-for-connection-strings-to-storage-accounts) instead of connectionstring will make it impossible to retry messages from service control.
+* Leveraging [multiple storage accounts](/nservicebus/azure-storage-queues/multi-storageaccount-support.md) will require a setup of multiple service control instances.
+* Choosing a [non default serializer for the message wrapper](/nservicebus/azure-storage-queues/configuration.md#configuration-parameters-serializemessagewrapperwith) or using a [custom envelopewrapper](/nservicebus/azure-storage-queues/configuration.md#custom-envelope-unwrapper) may lead to incompatible wire formats.
```
Comment (SeanFeldman) [inline servicecontrol/transport-adapter/incompatible-features.md]: Same question, may or will

**Comment by SimonCropp** [comment]
would be great if a @Particular/servicecontrol-maintainers could review this as well

File: servicecontrol/transport-adapter/incompatible-features.md (hunk 2 above)
Comment (SimonCropp) [inline servicecontrol/transport-adapter/incompatible-features.md]: also ServiceControl is no space

File: nservicebus/azure-service-bus/brokered-message-creation.md [hunk 5]
```diff
@@ -25,7 +25,7 @@ NOTE: Both the sender (native or NServiceBus) and the receiver must have a consi
 
 By default, the `BrokeredMessage` body is assumed to be a byte array to remain backward compatible with previous versions of the transport. It is recommended that the body can be retrieved using `Stream` for new implementations, as it incurs less of a memory overhead during serialization. It is especially useful for scenarios, such as native integration, as most non .NET ASB SDK's only support the `Stream` body format.
 
-NOTE: The `SupportedBrokeredMessageBodyTypes.Stream` body format is recommended, but this mode is currently NOT compatible with ServiceControl.
+NOTE: The `SupportedBrokeredMessageBodyTypes.Stream` body format is recommended, but this mode is currently NOT compatible with ServiceControl. A [ServiceControl transport adapter](/servicecontrol/transport-adapter/) is required in order to leverage both.
```
Comment (mikeminutillo) [inline nservicebus/azure-service-bus/brokered-message-creation.md]: Do we have a transport adapter for ASB/ASQ yet?

File: nservicebus/azure-storage-queues/multi-storageaccount-support_aliases_asq_[7,].partial.md (hunk 1 above)
Comment (mikeminutillo) [inline nservicebus/azure-storage-queues/multi-storageaccount-support_aliases_asq_[7,].partial.md]: I found the first sentence difficult to read.

> The examples above use different default accounts' aliases to enable a coherent addressing

File: nservicebus/azure-service-bus/brokered-message-creation.md (hunk 5 above)
Comment (yvesgoeleven) [inline nservicebus/azure-service-bus/brokered-message-creation.md]: @mikeminutillo my understanding was that users need to create such an adapter themselves. There is an issue next in line in platdev to make a bootstrapper to make it simpler,https://github.com/Particular/PlatformDevelopment/issues/1288, is that what you're referring too?

**Comment by yvesgoeleven** [comment]
@mikeminutillo that would be a good idea, can you @Particular/servicecontrol-maintainers  create a separate issue for this?

File: servicecontrol/transport-adapter/incompatible-features.md (hunk 3 above)
Comment (yvesgoeleven) [inline servicecontrol/transport-adapter/incompatible-features.md]: may, depends on what user does differently in custom impl

File: servicecontrol/transport-adapter/incompatible-features.md (hunk 4 above)
Comment (yvesgoeleven) [inline servicecontrol/transport-adapter/incompatible-features.md]: same answer :D

File: nservicebus/azure-storage-queues/multi-storageaccount-support_aliases_asq_[7,].partial.md (hunk 1 above)
Comment (yvesgoeleven) [inline nservicebus/azure-storage-queues/multi-storageaccount-support_aliases_asq_[7,].partial.md]: > NOTE: The examples above use different values for the default account aliases. Using the same name, such as `default`, to represent different storage accounts in different endpoints is highly discouraged as it introduces ambiguity in resolving addresses like `queue@default` and may cause issues when e.g. replying. In that case an address is interpreted as a reply address, the name `default` will point to a different connection string.

Is this more clear?

**Comment by MarcinHoppe** [comment]
@mikeminutillo @yvesgoeleven 👍 for a separate issue, where we could also address security. Our current strategy for security SC/SP is workable for on-premise deployment but quite problematic in the cloud.

**Comment by yvesgoeleven** [comment]
@MarcinHoppe as is scale https://github.com/Particular/PlatformDevelopment/issues/1330

**Comment by Scooletz** [comment]
LGTM from ASQ point of view.

**Comment by SeanFeldman** [comment]
👍 ASB is covered

**Comment by yvesgoeleven** [comment]
@SimonCropp @mikeminutillo you guys ok with this? if yes, feel free to merge.

**Pull request merged**